test/test.js

```diff
@@ -13064,6 +13064,29 @@
 
   /*--------------------------------------------------------------------------*/
 
+  QUnit.module('lodash.replaceAll');
+
+  (function() {
+    test('should replace all occurences of pattern in a string', 2, function() {
+      strictEqual(_.replaceAll('abc/def/ghi', '/', '-'), 'abc-def-ghi');
+      strictEqual(_.replaceAll('abc||def||ghi', '||', '.'), 'abc.def.ghi');
+    });
+
+    test('should replace nested patterns', 1, function() {
+      strictEqual(_.replaceAll('a/b/c', '/', '//'), 'a//b//c');
+    });
+
+    test('should coerce `pattern` and `replacement` to a string', 1, function() {
+      strictEqual(_.replaceAll('123456', 3, '4'), '124456');
+    });
+
+    test('should coerce `replacement` to a string', 1, function() {
+      strictEqual(_.replaceAll('123', '2', { 'toString': _.constant('4') }), '143');
+    });
```

Comment:
I was mentioning it for chain cases
On Apr 20, 2015 11:46 AM, "John-David Dalton" <notifications@github.com>
wrote:

> In test/test.js
> <https://github.com/lodash/lodash/pull/1123#discussion_r28700455>:
>
> > +    test('should replace all occurences of pattern in a string', 2, function() {
> > +      strictEqual(_.replaceAll('abc/def/ghi', '/', '-'), 'abc-def-ghi');
> > +      strictEqual(_.replaceAll('abc||def||ghi', '||', '.'), 'abc.def.ghi');
> > +    });
> > +
> > +    test('should replace nested patterns', 1, function() {
> > +      strictEqual(_.replaceAll('a/b/c', '/', '//'), 'a//b//c');
> > +    });
> > +
> > +    test('should coerce `pattern` and `replacement` to a string', 1, function() {
> > +      strictEqual(_.replaceAll('123456', 3, '4'), '124456');
> > +    });
> > +
> > +    test('should coerce `replacement` to a string', 1, function() {
> > +      strictEqual(_.replaceAll('123', '2', { 'toString': _.constant('4') }), '143');
> > +    });
>
> The param specifies it as a string, I'd assume if someone wanted regexes
> they'd use String#replace.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lodash/lodash/pull/1123/files#r28700455>.
>